### PR TITLE
Introduce more metrics for JS File Lookup

### PR DESCRIPTION
**Pull request by Swatinem**
In order to better measure the success of JS symbolication, this introduces more fine grained metrics that represent a funnel:

We capture metrics for source files and sourcemaps separately, with the following tags:
- found via debug-id
- found via release
- found via release-old
- found via scraping
- not found
- not needed (for sourcemaps)

This reuses existing metrics names as much as possible for now.

Additionally, this also captures more metrics for engineers that go into detail via which method the bundle was found.

fixes https://github.com/getsentry/sentry/issues/49950
#skip-changelog

### CODE REVIEW

File: crates/symbolicator-service/src/services/sourcemap_lookup.rs [hunk 1]
```diff
@@ -708,6 +696,14 @@ impl ArtifactFetcher {
             }
         };
 
+        if matches!(sourcemap.uri, CachedFileUri::Embedded) {
+            self.metrics
+                .record_not_found(SourceFileType::SourceMap, false);
```
Comment (loewenheim) [inline crates/symbolicator-service/src/services/sourcemap_lookup.rs:701]: This records embedded sourcemaps as not needed, right? Might it make more sense to record embedded sourcemaps as found?

File: crates/symbolicator-service/src/types/mod.rs [hunk 2]
```diff
@@ -704,14 +704,32 @@ pub struct JsFrameData {
     pub symbolicated: bool,
 }
 
-#[derive(Debug, Clone, Copy, Deserialize, Serialize, PartialEq, Eq)]
+/// A marker what a File was resolved with.
```
Comment (loewenheim) [inline crates/symbolicator-service/src/types/mod.rs:707]:
```suggestion
/// A marker indicating what a File was resolved with.
```

File: crates/symbolicator-service/src/services/sourcemap_lookup.rs (hunk 1 above)
Comment (Swatinem) [reply crates/symbolicator-service/src/services/sourcemap_lookup.rs:701]: Hm, but you don’t need to *fetch* it from anywhere.

File: crates/symbolicator-service/src/services/sourcemap_lookup.rs [hunk 3]
```diff
@@ -785,23 +781,31 @@ impl ArtifactFetcher {
...
+                        key.as_type(),
+                        ResolvedWith::DebugId,
+                        ResolvedWith::BundleIndex,
+                    );
                     tracing::trace!(?key, "Found file in `BundleIndex` by debug-id");
                     return Some(CachedFileEntry {
                         uri: CachedFileUri::Bundled(bundle_uri.clone(), key.clone()),
                         entry: CachedFile::from_descriptor(key.abs_path(), descriptor),
-                        resolved_with: Some(ResolvedWith::Index),
+                        resolved_with: ResolvedWith::BundleIndex, // TODO: should this be `DebugId` instead?
```
[the remark this answers is not in the document]
Comment (Swatinem) [reply crates/symbolicator-service/src/services/sourcemap_lookup.rs:793]: looks like we have cross-repo tests that assert `index` here anyway, so we can’t change that as easily. We should be able to revisit this at a later time though.

File: crates/symbolicator-service/src/services/sourcemap_lookup.rs (hunk 1 above)
Comment (loewenheim) [reply crates/symbolicator-service/src/services/sourcemap_lookup.rs:701]: Yeah, that's also true. I think it's fine the way it is.

**Review by loewenheim — APPROVED** [review]

**Pull request merged**